### DISCUSSION TITLE
Keycloak 21 and migration plan for new store

### DISCUSSION THREAD

**Discussion by kkcmadhu** [keycloak/keycloak#17324] [Archive]
Hi, now that the new store is there for a while, is this store now production ready now?
Whats keycloak recommendation for postgres deployment?
Whats keycloak's recommendation for mysql deployments?
Also, is there some kind of migration tool for existing postgres and mysql based deployments?

**Comment by vramik** [comment]
Hello @kkcmadhu,

the new store is still an experimental feature, so not production ready yet. 
There are plans to introduce a migration tool to make the migration from the legacy store easy. 

You can keep an eye on https://www.keycloak.org/blog to see updates regarding the topic.

**Comment by mm-matthias** [comment]
Hello @vramik,

is there an update regarding the plans of the new storage? To persist our sessions and action tokens we've been using Infinispan with a JDBC store to write directly to a Postgres database in the past. Unfortunately this is now severely broken in recent Keycloak versions (c.f. https://issues.redhat.com/browse/ISPN-14848 and the corresponding zulip chat).
So we are really interested in switching to the new storage. Is there anything we can do to help, e.g. doing beta-testing? Or is it too early at the moment and we should be more patient?

**Reply by stianst** [reply]
Hi @mm-matthias 

We don't have any updates on the new store right now. However, based on the linked Zulip chat it seems that you've found a solution to your problem and it's again working with persisting sessions through an external Infinispan server?

**Reply by mm-matthias** [reply]
Hi @stianst, thanks for your message. It's not exactly fixed as we ran into a few more bugs in Keycloak like https://github.com/keycloak/keycloak/pull/15619 and topologyId=-1 warnings which we are backporting to KC 20 at the moment. So in the long run, we'd love to migrate to a simplified (and hopefully more stable) storage, but for now we have the most pressing problem solved.
